Tests/SonarScanner.MSBuild.Tasks.IntegrationTests/MSBuildExecution/BuildLogAssertions.cs

```diff
@@ -0,0 +1,177 @@
+                propertyValue.Should().Be(expectedValue, "Property '{0}' does not have the expected value", propertyName);
+            }
+        }
+
+        public static string AssertPropertyExists(this BuildLog log, string propertyName)
+        {
+            var exists = log.TryGetPropertyValue(propertyName, out var propertyValue);
+            exists.Should().BeTrue(
+                "Expecting the property to exist. Property: {0}, Value: {1}", propertyName, propertyValue);
+
+            return propertyValue;
+        }
+
+        public static void AssertPropertyDoesNotExist(this BuildLog log, string propertyName)
+        {
+            var exists = log.TryGetPropertyValue(propertyName, out var propertyValue);
+            exists.Should().BeFalse(
+                "Not expecting the property to exist. Property: {0}, Value: {1}", propertyName, propertyValue);
+        }
+
+        public static void AssertTargetExecuted(this BuildLog log, string targetName)
+        {
+            var found = log.Targets.FirstOrDefault(t => t.Equals(targetName, StringComparison.InvariantCulture));
+            found.Should().NotBeNull("Specified target was not executed: {0}", targetName);
+        }
+
+        public static void AssertTargetNotExecuted(this BuildLog log, string targetName)
+        {
+            var found = log.Targets.FirstOrDefault(t => t.Equals(targetName, StringComparison.InvariantCulture));
+            found.Should().BeNull("Not expecting the target to have been executed: {0}", targetName);
+        }
+
+        public static void AssertTaskExecuted(this BuildLog log, string taskName)
+        {
+            var found = log.Tasks.FirstOrDefault(t => t.Equals(taskName, StringComparison.InvariantCulture));
+            found.Should().NotBeNull("Specified task was not executed: {0}", taskName);
+        }
+
+        public static void AssertTaskNotExecuted(this BuildLog log, string taskName)
+        {
+            var found = log.Tasks.FirstOrDefault(t => t.Equals(taskName, StringComparison.InvariantCulture));
+            found.Should().BeNull("Not expecting the task to have been executed: {0}", taskName);
+        }
+
+        /// <summary>
+        /// Checks that the expected tasks were executed in the specified order
+        /// </summary>
+        public static void AssertExpectedTargetOrdering(this BuildLog log, params string[] expected)
+        {
+            foreach (var target in expected)
+            {
+                AssertTargetExecuted(log,target);
+            }
+
+            var actual = log.Targets.Where(t => expected.Contains(t, StringComparer.Ordinal)).ToArray();
+
+            Console.WriteLine("Expected target order: {0}", string.Join(", ", expected));
+            Console.WriteLine("Actual target order: {0}", string.Join(", ", actual));
+
+            CollectionAssert.AreEqual(expected, actual, "Targets were not executed in the expected order");
```

Comment:
There is a collection assertion in FluentAssertion so I would use this one instead.

Reply:
I didn't bother previously because the default behaviour of the collection assertion in FA is to ignore ordering and working out how to get the equivalent behaviour wasn't a priority.
For reference, it's _actual.ShouldBeEquivalentTo(expected, options => options.WithStrictOrdering());_